File: tools/node/validate/validate_input.js
#!/usr/bin/env node
/**
 * Validate an AOM input surface (aom-input-schema.json).
 *
 * Usage:
 *   node Tools/node/validate/validate_input.js <aom-json-path>
 */

import path from "path";
import { fileURLToPath } from "url";
import { spawnSync } from "child_process";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const REPO_ROOT = path.resolve(__dirname, "..", "..", "..");
const CORE_VALIDATOR = path.join(__dirname, "validate.js");
const SCHEMA = path.join(REPO_ROOT, "spec", "v0.1.0", "aom-input-schema.json");

function main() {
  const args = process.argv.slice(2);
  if (!args[0]) {
    console.error("Usage: node Tools/node/validate/validate_input.js <aom-json-path>");
    process.exit(1);
  }

  const jsonPath = path.resolve(args[0]);
  const result = spawnSync(process.execPath, [CORE_VALIDATOR, SCHEMA, jsonPath], {
    cwd: REPO_ROOT,
    stdio: "inherit",
  });
  process.exit(result.status ?? 1);
}

main();

File: tools/python/validate/validate_input.py
#!/usr/bin/env python3
"""
Validate an AOM input surface (aom-input-schema.json).

Usage:
  python tools/python/validate/validate_input.py <aom-json-path>
"""

from pathlib import Path
import sys

from validate import main as _core_main  # type: ignore

REPO_ROOT = Path(__file__).resolve().parent.parent.parent.parent
SCHEMA = REPO_ROOT / "spec" / "v0.1.0" / "aom-input-schema.json"


def main():
    if len(sys.argv) != 2:
        print("Usage: python tools/python/validate/validate_input.py <aom-json-path>")
        sys.exit(1)

    json_path = Path(sys.argv[1]).resolve()
    sys.argv = [sys.argv[0], str(SCHEMA), str(json_path)]
    _core_main()


if __name__ == "__main__":
    main()

